src/components/Pages/OfficialDocuments/OfficialDocumentPage.js

```diff
@@ -0,0 +1,179 @@
+import React, { useState, useEffect  } from 'react'
+import { injectIntl } from 'react-intl'
+import { navigation as i18n2 } from 'js/i18n/definitions';
+import { isMobileQuery } from 'js/helpers/reactMediaQueries.js';
+
+import ChevronRight from 'components/SVGs/ChevronRight'
+import ChevronLeftBlue from 'components/SVGs/ChevronLeftBlue'
+import ChevronRightBlue from 'components/SVGs/ChevronRightBlue'
+import OfficialDocument from 'components/Pages/OfficialDocuments/OfficialDocument'
+
+const OfficialDocumentPage = ({ officialDocuments, intl }) => {
+
+  const documentsPerPage = 10
+  const isMobile = isMobileQuery()
+  const maxPagesShown = isMobile ? 4 : 7 // Desktop : Mobile (pagination pages shown)
+  const allDocs = officialDocuments.edges
+  const pages = buildPages()
+  const [ pageNumber, setPageNumber ] = useState(getHash())
+  const shownPages = buildPagination()
+  const page = pages[pageNumber]
+
+  useEffect(() => {
+    const str = typeof window !== 'undefined' ? window.location.hash.split("#")[1] : 0
+    const hash = (str > 0 && str <= pages.length) ? parseInt(str)-1 : 0
+    setPageNumber(hash)
+  })
+
+  useEffect(() => {
+    window.onpopstate = function(event) {
+      setPageNumber(getHash())
+      window.requestAnimationFrame( ()=> window.scroll(0,0) )
+    }
+    window.location.hash = pageNumber+1
+  })
+
+  function getHash(){
+    const str = typeof window !== 'undefined' ? window.location.hash.split("#")[1] : 0
+    const hash = (str > 0 && str <= pages.length) ? parseInt(str)-1 : 0
+    return hash
+  }
+
+  function buildPages() {
+    const pages = []
+    for (var i = 0; i < allDocs.length; i += documentsPerPage) {
+      pages.push(allDocs.slice(i, i + documentsPerPage))
+    }
+    return pages
+  }
+
+  function buildPagination() {
```

Comment:
The way the truncation in the pagination works... 
- First the `maxPagesShown` variable sets the maximum amount of displayed pages within the pagination. 
- Then, the *1st* and *last* pages are shown.
- If there are 3 or more pages show, the current page is shown. 
- If there are 4 or more pages shown, we add the next page after the shown page, then the previous page. 
  - That process repeats(adding1) until all the pages are filled, or the ranges are fill. 
- THEN, the gaps between pages are filled with "..."